### PR TITLE
[Bugfix] scores csv broken with duplicate title

### LINKED ISSUE #2818: [Bug] Exam Scores, Course Scores Export broken when duplicated exercise group or exercise title
### Describe the bug

The CSV export of course scores is broken when two exercises have the same title. (data is in the wrong column)
The CSV export of exam scores is broken when two exercise groups have the same title. (data is in the wrong column)

**The code's problem is that we map the exercise title or exercise group title to the data. If exercise A and exercise B have the same title, there will be only one entry in the map**

#### To Reproduce

- Go to the test course and change the names of exercises / exercise groups and test the CSV export function. Data will be in the wrong column

#### Expected behaviour

CSV export should gracefully handle the case when two exercises and exercise groups have the same title

### PR DESCRIPTION
**Pull request by stefanwaldhauser**
### Checklist
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)
- [x] Client: I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context

Hotfix that closes #2818

### Description

Add the id to duplicated titles of exercise groups and exercises to make the titles unique

### Steps for Testing

**Test the course scores and exam scores page for courses and exam with exercises and exercise groups with duplicated title**

### Test Coverage

No changes

### CODE REVIEW

**Review by ivanchimeno — APPROVED** [review]
It works for me on TS1!

**Review by FrankeLukas — DISMISSED** [review]
Works as advertised.

**Review by FuchsDominik — DISMISSED** [review]
Testing worked well :)

File: src/main/webapp/app/course/course-scores/course-scores.component.ts [hunk 1]
```diff
@@ -81,6 +81,26 @@ export class CourseScoresComponent implements OnInit, OnDestroy {
         this.paramSub = this.route.params.subscribe((params) => {
             this.courseService.findWithExercises(params['courseId']).subscribe((res) => {
                 this.course = res.body!;
+
+                const titleMap = new Map();
```
Comment (krusche) [inline src/main/webapp/app/course/course-scores/course-scores.component.ts:85]: can you add the type annotations <string, number> for key+value for the map?

File: src/main/webapp/app/exam/exam-scores/exam-scores.component.ts [hunk 2]
```diff
@@ -80,6 +80,25 @@ export class ExamScoresComponent implements OnInit, OnDestroy {
                     if (this.examScoreDTO) {
                         this.studentResults = this.examScoreDTO.studentResults;
                         this.exerciseGroups = this.examScoreDTO.exerciseGroups;
+
+                        const titleMap = new Map();
```
Comment (krusche) [inline src/main/webapp/app/exam/exam-scores/exam-scores.component.ts:84]: can you add the type annotations <string, number> for key+value for the map?

**Review by krusche — CHANGES_REQUESTED** [review]
the code looks good, even if I am not a big fan of the code duplication and the missing tests ;-)

I still would like though that we add type annotations for the map datatypes so that we get more compiler safety

**Review by JohannesWeiss — APPROVED** [review]
Tested on ts1 and works as expected now, id is included for duplicates.

**Review by airelawaleria — APPROVED** [review]

**Review by EneaGore — DISMISSED** [review]

**Review by simonzilker — COMMENTED** [review]
looks good on ts1

**Review by JohannesWeiss — APPROVED** [review]
Re-Approve

**Review by airelawaleria — APPROVED** [review]

**Review by nicolasruscher — DISMISSED** [review]
Code lgtm 👍

**Review by marvinchimeka — DISMISSED** [review]
Works as intended. Tested on ts1

**Review by FrankeLukas — DISMISSED** [review]
Re-approve

**Review by simonzilker — DISMISSED** [review]

**Review by valeriebucher — DISMISSED** [review]

**Review by philgabr — DISMISSED** [review]
works as described :+1:

**Review by krusche — DISMISSED** [review]

**Review by krusche — APPROVED** [review]

**Pull request merged**